remote_execution/oss/re_grpc/src/client.rs

```diff
@@ -559,8 +502,7 @@ impl REClient {
     }
 
     pub fn get_network_stats(&self) -> anyhow::Result<NetworkStatisticsResponse> {
-        // TODO(aloiscochard): Is that unwrap safe? could we avoid it by making the caller async?
-        let state = self.state.lock().unwrap();
+        let state = self.state.lock().unwrap_or_else(|e| e.into_inner());
```

Comment:
What does this do? The error goes into_inner? I think just unwrap here is what we'd recommend, no "or_else"

Reply:
As far as I understand, this code means that if the locking fails, we take the data and return it anyway (https://doc.rust-lang.org/std/sync/struct.Mutex.html#method.into_inner).

This is probably the best thing we should do (instead of crashing), as we are only reading the state in this case and not writing to it.

_Please note that change was suggested during previous code review feedback._